core/src/rpc.rs

```diff
@@ -409,21 +407,26 @@ impl JsonRpcRequestProcessor {
         let bank = self.bank(commitment);
 
         for signature in signatures {
-            let status = bank.get_signature_confirmation_status(&signature).map(
-                |SignatureConfirmationStatus {
-                     slot,
-                     status,
-                     confirmations,
-                 }| TransactionStatus {
-                    slot,
-                    status,
-                    confirmations: if confirmations <= MAX_LOCKOUT_HISTORY {
-                        Some(confirmations)
-                    } else {
+            let status = bank
+                .get_signature_status_slot(&signature)
+                .map(|(slot, status)| {
+                    let r_block_commitment_cache = self.block_commitment_cache.read().unwrap();
+
+                    let confirmations = if r_block_commitment_cache.root() >= slot {
```

Comment:
Hmmm do we not care about signatures from before the last root?

Reply:
Sure we do. We just don't care about the number of confirmations form before the last root.

Reply:
Hmm, can't there be rooted slots < last root with signatures that have MAX_CONFIRMATIONS? Recall that the history of queryable roots in the status cache is much longer than the last root

Reply:
Yes, exactly. We are returning `None` here for any signature in the last root, or any older rooted slot